libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c

```diff
@@ -2399,6 +2404,42 @@ static void prvAgentShutdownCleanup( void )
     }
 }
 
+/*
+  * Handle any events that were unexpected in the current state.
+  */
+static void prvHandleUnexpectedEvents( OTA_EventMsg_t * pxEventMsg )
+{
+    DEFINE_OTA_METHOD_NAME( "prvHandleUnexpectedEvents" );
+
+    OTA_LOG_L1( "[%s] Unexpected Event. Current State [%s] Received Event  [%s] \n",
+                OTA_METHOD_NAME,
+                pcOTA_AgentState_Strings[ xOTA_Agent.eState ],
+                pcOTA_Event_Strings[ pxEventMsg->xEventId ] );
+
+    /* Perform any cleanup operations required for specifc unhandled events.*/
+    switch( pxEventMsg->xEventId )
+    {
+        case eOTA_AgentEvent_ReceivedJobDocument:
+
+            /* Received job event is not handled , release the buffer.*/
+            prvOTAEventBufferFree( pxEventMsg->pxEventData );
+
+            break;
+
+        case eOTA_AgentEvent_ReceivedFileBlock:
+
+            /* Received file data event is not handled , release the buffer.*/
+            prvOTAEventBufferFree( pxEventMsg->pxEventData );
+
+            break;
+
+        default:
+
+            /* Nothing to do here.*/
```

Comment:
Wouldn't you always want to log here?  It's an unhandled unexpected event, and logging could be disabled on line 2414, correct?

Reply:
On line 2414 we are logging all the unhandled events and what state OTA Agent is, then for some events we do required cleanup. So all the events are anyway unexpected event, moving the logging to default will require us to add logging other cases as well , but with same information.